networkx/algorithms/hierarchy_coordinates.py

```diff
@@ -0,0 +1,1107 @@
+    >>> condensed_networks, _ = nx.node_weighted_condense(b, num_thresholds=num_thresholds)
+    >>> for cG in condensed_networks:
+    ...     for node, weight in cG.nodes.data("weight"):
+    ...         print(f"Node {node}, new weight: {weight}")
+    Node 0, new weight: 1
+    Node 1, new weight: 3
+    Node 2, new weight: 1
+    Node 0, new weight: 1
+    Node 1, new weight: 1
+    Node 2, new weight: 1
+
+    Raises
+    ------
+    NetworkXError
+        If input not a directed acyclic graph
+
+    Note:
+    ------
+    TODO: Might wish to eliminate return, or enable copying?
+
+    .. [1] "On the origins of hierarchy in complex networks."
+     Corominas-Murtra, Bernat, Joaquín Goñi, Ricard V. Solé, and Carlos Rodríguez-Caso,
+     Proceedings of the National Academy of Sciences 110, no. 33 (2013)
+
+    See Also
+    --------
+    node_weighted_condense, graph_entropy
+    """
+    if not nx.is_directed_acyclic_graph(condensed_graph):
+        raise nx.NetworkXError(
+            "G must be a directed acyclic graph for node weighted condensation"
+        )
+
+    for node, attrs in condensed_graph.nodes.data():
+        attrs["weight"] = len(attrs["members"])
+    # WIP TODO: May not be necessary, as the graph itself is updated (not copied)?
+    return condensed_graph
+
+
+@not_implemented_for("undirected")
+def max_min_layers(G, max_layer=True):
+    """Returns the maximal (k_in = 0) layer or the minimal layer (k_out = 0)
+
+    Returns the maximal (k_in = 0, highest in hierarchy) layer (those nodes with in degree = 0)
+     or the minimal layer (k_out = 0) of a directed network.
+
+    Parameters
+    ----------
+    G: NetworkX Graph
+        A directed graph.
+    max_layer: bool, default: True
+        if True, returns maximal layer (k_in = 0), else returns nodes for which k_out = 0, minimal layer
+
+    Return
+    ------
+    min/max_layer: list
+        list of node indices as ints
+
+    Examples:
+    >>> import numpy as np
+    >>> a = np.array([
+    ...     [0, 0, 1, 1, 0, 0, 0],
+    ...     [0, 0, 1, 1, 0, 0, 0],
+    ...     [0, 0, 0, 0, 1, 1, 0],
+    ...     [0, 0, 0, 0, 0, 1, 1],
+    ...     [0, 0, 0, 0, 0, 0, 0],
+    ...     [0, 0, 0, 0, 0, 0, 0],
+    ...     [0, 0, 0, 0, 0, 0, 0],
+    ... ])
+
+    >>> G = nx.from_numpy_matrix(a, create_using=nx.DiGraph)
+    >>> print(nx.max_min_layers(G))
+    [0, 1]
+    >>> print(nx.max_min_layers(G, max_layer=False))
+    [4, 5, 6]
+
+    Raises
+    ------
+    NetworkXError
+        If input not a directed graph
+
+    Notes:
+    -------
+
+    """
+    if not G.is_directed():
+        raise nx.NetworkXError("G must be a DiGraph for min/max layer evaluation")
+    if max_layer:
+        return [node for node, deg in G.in_degree if deg == 0]
+    else:
+        return [node for node, deg in G.out_degree if deg == 0]
+
+
+@not_implemented_for("undirected")
+def leaf_removal(G, top=True):
+    """Returns a pruned network, with either maximal (k_in=0)
```

Comment:
Tighten up this language.  And add a paragraph to describe it more fully. What do you mean by pruned? Say something about why this function is useful, or how it fits wit the other code around it -- when should users use it?